plugins/woocommerce/src/Internal/ProductFilters/FilterData.php

```diff
@@ -373,6 +374,10 @@ private function get_cache( $key ) {
 	 * @param mix    $value Value to set.
 	 */
 	private function set_cache( $key, $value ) {
+		if ( ! is_array( $value ) ) {
+			return;
+		}
+
 		$transient_version = WC_Cache_Helper::get_transient_version( CacheController::TRANSIENT_GROUP );
 		$transient_value   = array(
 			'version' => $transient_version,
```

Comment:
I checked the class and what do you think about improving the `set_cache` docblock?

FROM:
```
	/**
	 * Set the cache with transient version to invalidate all at once when needed.
	 *
	 * @param string $key   Transient key.
	 * @param mix    $value Value to set.
	 */
```

TO:

```
/**
 * Set the cache with transient version to invalidate all at once when needed.
 *
 * @param string $key   Transient key.
 * @param array  $value Value to set.
 * @return bool|null Returns true if transient was set successfully, false on failure, or null if value is not an array.
 */
```

Reply:
Makes sense! I updated the first early return to return `false`, making this method only returns boolean.

Reply:
Nice one! 😄 You read my mind, I wanted to suggest it but I thought it's gonna be unreadable if I start pointing out things not covered in the PR. But you changed that anyway, thank you! 🚀